## Data Acquisition with SQL Project

## Data Analysis Question

#### The context

Our task is to understand car crashes in the city of Chicago. As part of the descriptive analysis report, we will find answers to the following questions.

> 
1. Display a list of 5 makes that had the highest number of crashes?
2. Most personal vehicles involved in crashes were traveling towards which direction?
3. How many passengers were involved in personal cars were in accidents?
4. How many passengers were involved in accidents while traveling in personal Sport
Utility vehicles?
5. Compute the total number of Mercedez-Benz vehicles involved in accidents while
being parked.
6. Compute the total occupant count per vehicle count.
7. Sort the number of passengers in descending order by travel direction taking into
account travel direction. Return travel direction and no. of passengers.
8. Select the top 5 vehicle models involved in the accident with no defect, order them
by make, and show their maneuver.
9. Which ridesharing vehicles in an accident had the most number of passengers?
Return model, and the number of passengers.


#### Solution steps

In order to answer the questions above, we will;
1. Connect to the database
2. Load libraries and the dataset
3. Create database table and preview data
4. Perform data analysis
5. Summarise our findings

## Connect to the database

Load the SQL extension

In [1]:
%load_ext sql

Connect to the in-memory SQLite database within Google Colab

In [2]:
%sql sqlite://

'Connected: @None'

## Load libraries and the dataset

Load libraries

In [3]:
import pandas as pd

Load data from CSV

In [4]:
traffic_accidents = pd.read_csv('/content/traffic_accidents.csv')

## Create database table and preview data

Create database table using the loaded CSV data

In [5]:
%sql DROP TABLE if EXISTS traffic_accidents;

%sql PERSIST traffic_accidents

 * sqlite://
Done.
 * sqlite://


'Persisted traffic_accidents'

Preview the data

In [6]:
%sql SELECT * FROM traffic_accidents LIMIT 10

 * sqlite://
Done.


[(0, 54, '0ecbc94e9e6a01f8d1fadca458d84be6d51ef9d42b1f4c2f4367a2388fef03451e8bcd7a797f7dfb84f03cf362f81e56429ac15787893f6350e7166d54ad9cff', 'HY372069', '08/06/2015 06:00:00 PM', 2, 'PARKED', None, 53.0, None, 'LINCOLN-CONTINENTAL', 'MKX', 'IL', 2008.0, 'NONE', 'PASSENGER', 'NOT IN USE', 'N', 'PARKED', None, None, 0.0, None, None, None, None, None, None, 'Y', 'Y', None, None, None, None, None, None, None, None, None, 'REAR-RIGHT', None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None),
 (1, 145, '7bd0f17b2a365cb53396bf29ac78123c1c258465aa0f1773c792a03d1b36f98e92f60a8ebb87ebaee5f957a3e9813b9aa5ac4f6430498bc0e3a64167765e7165', 'HY376728', '08/10/2015 02:15:00 PM', 2, 'DRIVER', None, 141.0, None, 'MERCEDES-BENZ', 'C250', 'IL', 2013.0, 'NONE', 'PASSENGER', 'PERSONAL', 'S', 'STRAIGHT AHEAD', None, None, 1.0, None, None, None, None, None, None, None, None, 'Y', None, None, None, None, None, None, None, None, 'REAR', None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None),
 (2, 257, '07155af1c0a71c36e32ab53f86b525d929154b3d7290ae0830b7baa834c7c6f2b22e14a6656406ccad8685bf5f9ac54b058dce68d7122bd468729a941f1c3a97', 'HY383388', '08/15/2015 12:10:00 PM', 1, 'DRIVER', None, 247.0, None, 'MERCEDES-BENZ', 'GLK220', 'IL', 2015.0, 'NONE', 'PASSENGER', 'OTHER', 'E', 'STRAIGHT AHEAD', None, None, 1.0, None, None, None, 'N', 'Y', None, None, None, None, None, None, None, None, None, None, None, None, 'FRONT', None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None),
 (3, 283, '6a54948070f52dcfc8682fc14a374871b76561244c2383fcb89004ebe418262f56c7249c1dfe00f1206dc7d8017d39da4ca23015e6739c668e12cddd302038ee', 'HY385182', '08/16/2015 07:00:00 PM', 2, 'DRIVER', 5.0, 270.0, None, 'TOYOTA MOTOR COMPANY, LTD.', 'CAMRY', 'IL', 2014.0, 'NONE', 'PASSENGER', 'PERSONAL', 'W', 'SLOW/STOP - RIGHT TURN', None, None, 6.0, None, None, None, None, 'Y', 'Y', None, None, None, None, None, 'Y', None, None, None, None, None, 'FRONT-LEFT', None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None),
 (4, 327, '0ff295104810bc4e8d6c15e4fc92a5beb3768c792bb995b80bf2fe3904e6d30390fce363ae8b4c99a0c48cca42697d43a385b8b654cfea77d59a7c11342fc664', 'HY386474', '08/16/2015 01:45:00 PM', 2, 'PARKED', None, 312.0, None, 'LEXUS', 'RX350', 'IL', 2015.0, 'NONE', 'SPORT UTILITY VEHICLE (SUV)', 'OTHER', 'N', 'PARKED', None, None, 0.0, None, None, None, None, None, None, None, 'N', 'Y', 'N', None, None, None, None, None, None, None, 'REAR', None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None),
 (5, 456, '497ee0da1507d49c1d13b5544c6d29accb42135494cd93925f0557f0e8f9513c9ca0de862ae12c9aa011106a8b2b4b548b29fefcf1773a571ea5f00262068a2c', 'HY390815', '08/20/2015 07:00:00 PM', 2, 'DRIVER', None, 435.0, None, 'FORD', 'FORD-TRUCK,VAN,SUV, ETC', 'IL', 2010.0, 'NONE', 'VAN/MINI-VAN', 'COMMERCIAL - MULTI-UNIT', 'W', 'STRAIGHT AHEAD', None, None, 1.0, None, None, None, None, None, None, None, None, 'Y', None, None, None, None, None, None, None, None, 'REAR', None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None),
 (6, 524, '7aa347e0ea02f41d7b3c50083a8b359145ace7f7dea28a535b7985d80c9a96bf6c17cb366cf08b9c17795064fea1505de042d5027a837fed5c7aef71f5432688', 'HY393132', '0

## Data analysis

**Question 1:** Display a list of 5 makes that had the highest number of crashes

In [8]:
%%sql
SELECT MAKE, count(*) as num_crashes FROM `traffic_accidents`  GROUP BY MAKE ORDER BY num_crashes DESC LIMIT 5

 * sqlite://
Done.


[('CHEVROLET', 726),
 ('FORD', 658),
 ('NISSAN', 585),
 ('TOYOTA MOTOR COMPANY, LTD.', 494),
 ('HONDA', 403)]

**Question 2:** Most personal vehicles involved in crashes were traveling towards which direction?

In [27]:
%%sql
SELECT TRAVEL_DIRECTION, count(*) as num_vehicles FROM `traffic_accidents` WHERE VEHICLE_USE = 'PERSONAL'  GROUP BY TRAVEL_DIRECTION ORDER BY num_vehicles DESC 

 * sqlite://
Done.


[('N', 960),
 ('S', 932),
 ('E', 854),
 ('W', 847),
 ('UNKNOWN', 149),
 ('SE', 65),
 ('NW', 63),
 ('SW', 39),
 ('NE', 32)]

Observation: 
> Most of the personal vehicles invoved in crashes were travelling towards North

**Question 3:** How many passengers were involved in personal cars were in accidents?

In [13]:
%%sql
SELECT sum(NUM_PASSENGERS) as num_of_passengers FROM `traffic_accidents` WHERE VEHICLE_USE = 'PERSONAL' 

 * sqlite://
Done.


[(1016.0,)]

**Question 4:** How many passengers were involved in accidents while traveling in personal Sport Utility vehicles?

In [14]:
%%sql
SELECT sum(NUM_PASSENGERS) as num_of_passengers FROM `traffic_accidents` WHERE VEHICLE_TYPE = 'SPORT UTILITY VEHICLE (SUV)' 

 * sqlite://
Done.


[(236.0,)]

**Question 5:** Compute the total number of Mercedez-Benz vehicles involved in accidents while being parked?

In [18]:
%%sql
SELECT count(*) as total FROM `traffic_accidents` WHERE MAKE = 'MERCEDES-BENZ' AND UNIT_TYPE = 'PARKED' 

 * sqlite://
Done.


[(20,)]

**Question 6:** Compute the total occupant count per vehicle type

In [24]:
%%sql
SELECT VEHICLE_TYPE, sum(OCCUPANT_CNT) as total FROM `traffic_accidents`  GROUP BY VEHICLE_TYPE ORDER BY total DESC

 * sqlite://
Done.


[('PASSENGER', 4150.0),
 ('SPORT UTILITY VEHICLE (SUV)', 926.0),
 ('VAN/MINI-VAN', 299.0),
 ('PICKUP', 148.0),
 ('UNKNOWN/NA', 123.0),
 ('BUS OVER 15 PASS.', 111.0),
 ('TRUCK - SINGLE UNIT', 96.0),
 ('OTHER', 76.0),
 ('TRACTOR W/ SEMI-TRAILER', 55.0),
 ('BUS UP TO 15 PASS.', 24.0),
 ('MOTORCYCLE (OVER 150CC)', 14.0),
 ('OTHER VEHICLE WITH TRAILER', 8.0),
 ('SINGLE UNIT TRUCK WITH TRAILER', 5.0),
 ('TRACTOR W/O SEMI-TRAILER', 4.0),
 ('ALL-TERRAIN VEHICLE (ATV)', 1.0),
 ('AUTOCYCLE', 1.0),
 ('MOPED OR MOTORIZED BICYCLE', 1.0),
 ('MOTOR DRIVEN CYCLE', 1.0)]

**Question 7:** Sort the number of passengers in descending order by travel direction taking into account travel direction. Return travel direction and no. of passengers.

In [25]:
%%sql
SELECT TRAVEL_DIRECTION, sum(NUM_PASSENGERS) as num_of_passengers FROM `traffic_accidents` GROUP BY TRAVEL_DIRECTION ORDER BY num_of_passengers DESC 

 * sqlite://
Done.


[('N', 371.0),
 ('S', 331.0),
 ('E', 305.0),
 ('W', 285.0),
 ('UNKNOWN', 35.0),
 ('NW', 28.0),
 ('SE', 20.0),
 ('NE', 18.0),
 ('SW', 10.0)]

**Question 8:** Select the top 5 vehicle models involved in the accident with no defect, order them by make, and show their maneuver.

In [46]:
%%sql
SELECT MODEL, MANEUVER FROM `traffic_accidents` WHERE VEHICLE_DEFECT = 'NONE' ORDER BY MAKE LIMIT 5;

 * sqlite://
Done.


[('"ALLISON\'S FIBERGLASS MFG., INC (PORT ORANGE., FLORIDA)."', 'TURNING LEFT'),
 ('"CLASSIC MOTOR CARRIAGES, INC. (HALLANDALE, FLORIDA)"', 'PASSING/OVERTAKING'),
 ('ACURA', 'STRAIGHT AHEAD'),
 ('ACURA', 'STARTING IN TRAFFIC'),
 ('ACURA', 'STRAIGHT AHEAD')]

**Question 9:** Which ridesharing vehicles in an accident had the most number of passengers? Return model, and the number of passengers

In [43]:
%%sql
SELECT MODEL, sum(NUM_PASSENGERS) as NUM_OF_PASSENGERS FROM `traffic_accidents` WHERE VEHICLE_USE = 'RIDESHARE SERVICE' GROUP BY MODEL ORDER BY NUM_OF_PASSENGERS DESC

 * sqlite://
Done.


[('CAMRY', 5.0),
 ('UNKNOWN', 5.0),
 ('JETTA', 4.0),
 ('COROLLA', 3.0),
 ('A4', 2.0),
 ('MALIBU (CHEVELLE)', 2.0),
 ('ACURA', 1.0),
 ('CX-9', 1.0),
 ('ELANTRA', 1.0),
 ('FLEX', 1.0),
 ('MKS', 1.0),
 ('OTHER (EXPLAIN IN NARRATIVE)', 1.0),
 ('PATRIOT', 1.0),
 ('SOUL', 1.0),
 ('Trail Blazer', 1.0),
 ('VERSA', 1.0),
 ('328i', None),
 ('ALTIMA', None),
 ('CX-3', None),
 ('EQUINOX', None),
 ('ESCAPE (SUV)', None),
 ('FOCUS', None),
 ('FUSION', None),
 ('Highlander(beginning vehicle year 2001)', None),
 ('LINCOLN PARK MOBILE HOMES, INC., SHIPSHEWNAN, IN', None),
 ('MDX', None),
 ('NISSAN ALTIMA', None),
 ('OPTIMA / K5', None),
 ('PASSAT', None),
 ('Prius', None),
 ('RANGE ROVER OF NORTH AMERICA', None),
 ('SE-V6', None),
 ('SENTRA (DATSUN AND NISSAN HAVE MERGED)', None),
 ('SONATA', None),
 ('Sportage (sport utility)', None)]

Observation:
> CAMRY rideshare vehicles had the most passengers

## Summary of findings

From the analysis above, these are the findings

1. CHEVROLET vehicles had the highest number of crashes
2. Most personal vehicles were involved in accidents while travelling towards North
3. 1,016 passengers were in personal cars when the cars got accidents
4. Passengers travelling in Sport Utility vehicles totalled 236
5. 20 Mercedez-Benz vehicles were involved in accidents while being parked
6. PASSENGER vehicles had the most occupant count
7. AMRY rideshare vehicles had the most passengers